### ISSUE TITLE
Validator fails unexpressively when $ref appears in invalid location

### ISSUE THREAD

**Issue by tbenda-SEL** [IBM/openapi-validator#100]
Using `lint-openapi` version `0.13.3`

When a ref appears as a direct child of either the `paths` element or a given path in the `paths` element (two illegal positions) rather than failing with a useful error message, it exits with:

> [Error] There was a problem with a validator.
> Cannot read property '$ref' of undefined

Minimal reproduction pair here:

derp.yaml
```yaml
---
openapi: 3.0.0
info:
  version: 0.0.1
  title: derp
  contact:
    name: yeah
    email: nobody@example.com

paths:
  /derp:
    post:
      $ref: yarp.yaml#/derp-post
```

yarp.yaml
```yaml
---
derp-post:
  operationId: derp
  summary: derp
  requestBody:
    description: derp
    content:
      application/json:
        schema:
          type: object
  responses:
    '200':
      description: derp
```

It would be more helpful if it failed with a message reflecting that the location was illegal, though I understand these edge cases are really hard to catch. :smiley:

**Comment by dpopp07** [comment]
@tbenda-SEL I agree, the validator should be able to catch that as an error, report in, and continue with validation. I'll look into it, thanks for reporting
